util/git/ssh.go

```diff
@@ -11,13 +11,12 @@ import (
 // Unfortunately, crypto/ssh does not offer public constants or list for
 // this.
 var SupportedSSHKeyExchangeAlgorithms = []string{
```

Comment:
@34fathombelow pointed out that this list should probably be ordered. The crypto/ssh `findCommon` function will pick the first item from this list that is also in the server's list. https://cs.opensource.google/go/x/crypto/+/master:ssh/common.go;l=132?q=findcommon&ss=go%2Fx%2Fcrypto

Suggested order:

```
// Source for strength estimates: https://www.ietf.org/archive/id/draft-ietf-curdle-ssh-kex-sha2-15.html
	"ecdh-sha2-nistp521", // 512 bits
	"ecdh-sha2-nistp384", // 192 bits
	"ecdh-sha2-nistp256", // 128 bits
	"curve25519-sha256@libssh.org", // 128 bits
	// The security strength of the FFC and the hash together will be the minimum of those two values.
	"diffie-hellman-group14-sha256", // 112 bits
	"diffie-hellman-group14-sha1", // 112 bits
	"diffie-hellman-group-exchange-sha256",
```

I couldn't find a source for the strength of `diffie-hellman-group-exchange-sha256`.

Reply:
I'm not a cryptography expert by any means.  I do believe some of the algorithms come with a performance cost or other complexities that might not scale.   I'll do my best to do some research. 

If anyone has an opinion on this, I'd like to hear it.

Reply:
That's fair. I'd really like to track down a crypto expert to review this change. I'll see what I can do.

Reply:
Updated the list with all currently [supported](https://cs.opensource.google/go/x/crypto/+/793ad666bf5ec61392092b27061be9618e4e219b:ssh/kex.go) kex-algos with the exception of `diffie-hellman-group1-sha1` & `diffie-hellman-group-exchange-sha1` from the golang.org/x/crypto/ssh package. They are now sorted according to the latest [OpenSSH](https://man.openbsd.org/sshd_config#KexAlgorithms) defaults. The OpenSSH defaults are near identical to Ubuntu:22.04 defaults.

I have ran all unit & e2e test with these changes against 22.04, 21.10, and 20.04